autoload/lsp/ui/vim/references.vim

```diff
@@ -221,5 +221,5 @@ function! lsp#ui#vim#references#jump(offset) abort
 
     " Jump
     let l:target = w:lsp_reference_positions[l:index][0:1]
-    silent exec 'normal! ' . l:target[0] . 'G0' . (l:target[1] == 1 ? '' : l:target[1]-1 . 'l')
+    call cursor(l:target[0], l:target[1])
```

Comment:
I'm not sure, but is this 1 origin, isn't it?

Reply:
The lnum and col that cursor() uses both start at 1: `call cursor(3,5)` will place the cursor on the fifth byte of the third line of the buffer. I've tested the :Lsp{Next,Previous}Reference commands with this change and they function correctly.

Reply:
@mattn @dhakimian any update on this?

Reply:
https://microsoft.github.io/language-server-protocol/specification

Language Server Specification explain the position is 0 based.

Reply:
Yes, but the positions being worked with at this point in the file have already been converted from 0-based to 1-based (see the range_to_position function in the same file). The original line here used `G` and `|`, which are both 1-based, and my changes are also 1-based. I have confirmed my changes function as expected, and if you pull from my branch and test the reference jumping functions, you can see for yourself that they work properly.

Reply:
`cursor()` does not accept 0. And the `target` variable is possibly be zero.

Reply:
Are you sure about that? `target` is populated from the first two elements of an element in `w:lsp_reference_positions`, which comes from `l:position_list`, which is populated by the `s:range_to_position` function. Reading through this function, I don't see any way that the values that end up in `target` could be less than 1. If you can provide a set of circumstances that would result in `target` having a 0 value, let me know. Also let me know if, using my proposed changes, you can reproduce any instance of the Lsp{Next,Previous}Reference commands moving the cursor anywhere but the start of a span of text highlighted with the LspReference highlight group.

Reply:
Ah, sorry. I didn't look the part.